### PR TITLE
DELETE Milestone Completed

### PR DESCRIPTION
**Pull request by seancrossettie**
## Description
Completed the DELETE milestone. When authenticated and logged in, user can delete items from menu, ingredients, staff, and reservations. The items delete on Firebase and the DOM is dynamically rendered after each item is deleted.

## Link to deployed app
https://development--michelin-star-restaurant.netlify.app/#

## Related Issue
https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/issues/31
https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/issues/29
https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/issues/24
https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/issues/23

## Motivation and Context
Required for MVP.

## How Can This Be Tested?
Fetch the branch
```
git fetch
git checkout development
```
Install and start webpack
```
npm install
npm start
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/javascripts/components/navbar.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ const navbar = () => {
     </button>
     <div class="collapse navbar-collapse" id="navbarSupportedContent">
       <div class="navbar-nav">
-        <a class="nav-link active" id="nav-menu" href="#">Menu<span class="sr-only">(current)</span></a>
+        <a class="nav-link active" id="nav-menu" href="#">Menu</a>
```
Comment (TrinityChristiana) [inline src/javascripts/components/navbar.js:9]: Nice!

File: src/javascripts/menu items/menuData.js [hunk 2]
```diff
@@ -5,10 +5,31 @@ import firebaseConfig from '../helpers/apiKeys';
...
 
-export default getMenuItems;
+// DELETE MENU ITEMS
+const deleteMenuItems = (firebasekey, uid) => new Promise((resolve, reject) => {
+  axios.delete(`${dbUrl}/menu/${firebasekey}.json`)
+    .then(() => getMenuItems(uid).then((menuArray) => resolve(menuArray)))
+    .catch((error) => reject(error));
+});
+
+// CREATE MENU ITEMS
```
Comment (drteresavasquez) [inline src/javascripts/menu items/menuData.js]: Is this code you plan to use? We do not like to see commented out code on the main/production branch, so remove this.

**Review by drteresavasquez — CHANGES_REQUESTED** [review]
On your delete reservations, there is a bug where the delete prompt shows twice if I logout and in again. Make sure you are removing the event listeners on log out. Make a ticket for that and keep moving forward.

GREAT JOB on the buttons not showing if the user is not logged in! It is a beautiful sight! 👍🏼

File: src/javascripts/ingredients/ingredientsData.js [hunk 3]
```diff
@@ -4,10 +4,19 @@ import firebaseConfig from '../helpers/apiKeys';
 // API Calls for Ingredients
```
Comment (TrinityChristiana) [inline src/javascripts/ingredients/ingredientsData.js]: I didn't notice this before, but all of your API calls should live within the helper folder.

File: src/javascripts/menu items/menu.js [hunk 4]
```diff
@@ -16,4 +16,24 @@ const showMenuItems = (array) => {
...
+  document.querySelector('#modal-container').innerHTML = '';
+
+  array.forEach((item) => {
+    document.querySelector('#view').innerHTML += `<div class="media">
+    <img src="${item.image}" class="mr-3" id="menuItemImage"alt="...">
+    <div class="media-body">
+      <h5 class="mt-0">${item.title}</h5>
+      <p>${item.description}</p>
+      <p>${item.price}</p>
+      <button type="button" class="btn btn-secondary" id="edit-menu-item--${item.firebasekey}">Edit</button>
```
Comment (TrinityChristiana) [inline src/javascripts/menu items/menu.js]: Remove this edit button. We don't want nonfunctional elements in the production. This is also considered working ahead.

File: src/javascripts/menu items/menuData.js [hunk 5]
```diff
@@ -5,10 +5,31 @@ import firebaseConfig from '../helpers/apiKeys';
```
Comment (TrinityChristiana) [inline src/javascripts/menu items/menuData.js]: API calls should live in the helper folder.

File: src/javascripts/menu items/menuData.js (hunk 2 above)
Comment (TrinityChristiana) [inline src/javascripts/menu items/menuData.js]: Remove this code. We don't want commented out code in production. This is also considered working ahead!

File: src/javascripts/reservations/reservationData.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import 'firebase/auth';
```
Comment (TrinityChristiana) [inline src/javascripts/reservations/reservationData.js]: We want API calls to live in the helper's folder.

File: src/javascripts/reservations/reservations.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+const showReservations = (array) => {
```
Comment (TrinityChristiana) [inline src/javascripts/reservations/reservations.js]: nit: This should really live in the components folder

File: src/javascripts/reservations/reservations.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    document.querySelector('#view').innerHTML += `
+    <ul class="list-group-res">
+      <li class="list-group-item"><strong>Last Name:</strong> ${item.name}</li>
+      <li class="list-group-item"><strong>Party Size:</strong> ${item.party_size}</li>
+      <li class="list-group-item"><strong>Reservation Date:</strong> ${item.date}</li>
+      <li class="list-group-item"><strong>Day of the Week:</strong> ${item.day_of_week}</li>
+      <li class="list-group-item"><strong>Reservation Time:</strong> ${item.time}</li>
+      <li class="list-group-item"><strong>Notes:</strong> ${item.notes}</li>
+    </ul>
+    <button class="btn btn-info" data-toggle="modal" data-target="#formModal" id="edit-res-btn--${item.firebaseKey}">Edit Reservation</button>
```
Comment (TrinityChristiana) [inline src/javascripts/reservations/reservations.js]: Let's remove this Edit button. We don't want nonfunctional elements in the production. This is also considered working ahead.

File: src/javascripts/staff/staffData.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import axios from 'axios';
```
Comment (TrinityChristiana) [inline src/javascripts/staff/staffData.js]: We want the API calls to live in the helpers folder

File: src/javascripts/ingredients/showIngredients.js [hunk 10]
```diff
@@ -9,4 +9,16 @@ const showIngredients = (array) => {
   });
 };
 
-export default showIngredients;
+const showUserIngredients = (array) => {
```
Comment (TrinityChristiana) [inline src/javascripts/ingredients/showIngredients.js]: nit: The naming of this function is confusing. It reads as if the function returns ingredients that belong to a user.

File: src/javascripts/menu items/forms/formModal.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
+
+const formModal = (modalTitle) => {
```
Comment (TrinityChristiana) [inline src/javascripts/menu items/forms/formModal.js]: nit: If you have a components folder, any function that just returns HTML should live within that folder. You can separate your components out by feature there.

File: src/javascripts/menu items/menu.js [hunk 12]
```diff
@@ -6,7 +6,7 @@ const showMenuItems = (array) => {
```
Comment (TrinityChristiana) [inline src/javascripts/menu items/menu.js]: nit: Same with this file. It really should I've in the components folder.

File: src/javascripts/menu items/menu.js [hunk 13]
```diff
@@ -16,4 +16,24 @@ const showMenuItems = (array) => {
   });
 };
 
-export default showMenuItems;
+const showUserMenuItems = (array) => {
```
Comment (TrinityChristiana) [inline src/javascripts/menu items/menu.js]: nit: The naming of this function is confusing. It reads as if the function returns menu items that belong to a user.

File: src/javascripts/reservations/reservations.js [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+      <li class="list-group-item"><strong>Party Size:</strong> ${item.party_size}</li>
+      <li class="list-group-item"><strong>Reservation Date:</strong> ${item.date}</li>
+      <li class="list-group-item"><strong>Day of the Week:</strong> ${item.day_of_week}</li>
+      <li class="list-group-item"><strong>Reservation Time:</strong> ${item.time}</li>
+      <li class="list-group-item"><strong>Notes:</strong> ${item.notes}</li>
+    </ul>`;
+  });
+};
+
+const showUserReservations = (array) => {
```
Comment (TrinityChristiana) [inline src/javascripts/reservations/reservations.js]: nit: The naming of this function is confusing. It reads as if the function returns reservations that belong to a user.

**Review by TrinityChristiana — CHANGES_REQUESTED** [review]
I see that there was a decision to move all the helper functions out of the helper folder. Let's move the API calls back. 

I also see the same was happening with the components folder. I won't block you all on moving the components over, but any function that returns HTML is a component and really should live there. Maybe while completing the next milestones that can be updated.

I also see some code that is for future milestones. Remember we don't want to work ahead. Let's remove that code out of here.

Anything prefixed with "nit:" is not a blocker to getting this PR approved.

**Comment by TrinityChristiana** [comment]
Thank you for making the updates! I just need one more code change. It's not in the PR, but now that I'm testing the login state I see that the Staff page still has an Update button. Once that's updated I will get this PR approved for you all so that you can move on to the next milestone!

[Update button is here](https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/blob/main/src/javascripts/components/showStaff.js#L18)

**Review by TrinityChristiana — APPROVED** [review]
Looks great! You all are approved to move on to the next milestone!

**Review by elliewalkersm — APPROVED** [review]

**Review by drteresavasquez — APPROVED** [review]

**Pull request merged**